Translate Preface to PTBR (#42)
JuliaIntro/JuliaIntroBR.jl: book/src/introduction.asciidoc

[preface]
== Prefácio

> Em janeiro de 2018 eu comecei a preparar um curso de programação voltado à estudantes sem experiência de programação. Eu queria usar o Julia, mas descobri que não existia livro com o objetivo de aprender a programar com Julia como a primeira linguagem de programação. Existem tutoriais muito bons que explicam os conceitos chave do Julia, mas nenhum deles dá atenção suficiente a como aprender a pensar como um programador.

Reviewer: ```suggestion
Em Janeiro de 2018 eu comecei a preparar um curso de programação voltado à estudantes sem experiência de programação. Eu queria usar o Julia, mas descobri que não existia livro com o objetivo de aprender a programar com Julia como a primeira linguagem de programação. Existem tutoriais muito bons que explicam os conceitos chave do Julia, mas nenhum deles dá atenção suficiente a como aprender a pensar como um programador.
```

Todo o material estava disponível em notebooks Jupyter em um repositório no Github. Depois que mandei uma mensagem ao site do Julia Discourse sobre o progresso do meu curso, o feedback foi fantástico. Um livro sobre conceitos básicos de programação com Julia sendo a primeira linguagem de programação era aparentemente um elo perdido no universo Julia. Eu contatei Allen para perguntar se eu poderia começar uma adaptação oficial do _Think Python_ para o Julia, e sua resposta foi imediata: “Vá em frente!” ele me colocou em contato com o seu editor da O'Reilly Media, e um ano depois, eu estava colocando os toques finais neste livro.

> Foi ((("Julia", "version of"))) uma jornada e tanto. Em agosto de 2018 o Julia v1.0 foi lançado, e como todos meus companheiros que programam em Julia, eu tive que fazer uma migração do código. Todos os exemplos nesse livro foram testados durante a conversão dos arquivos fonte para arquivos AsciiDoc compativeís com a O'Reilly. Tanto o conjunto de ferramentas e o código exemplo teve que ser compatibilizado com o Julia v1.0. Por sorte, não há aulas para dar em Agosto....

Reviewer: ```suggestion
Foi ((("Julia", "versão do"))) uma jornada e tanto. Em Agosto de 2018 o Julia v1.0 foi lançado, e como todos meus companheiros que programam em Julia, eu tive que fazer uma migração do código. Todos os exemplos nesse livro foram testados durante a conversão dos arquivos fonte para arquivos AsciiDoc compativeís com a O'Reilly. Tanto o conjunto de ferramentas e o código exemplo teve que ser compatibilizado com o Julia v1.0. Por sorte, não há aulas para dar em Agosto....
```

Eu espero que você aproveite trabalhar com este livro, e que ele o ajude a aprender a programar e a pensar como um cientista da computação, pelo menos um pouquinho.

[role="byline"]
Ben Lauwens

> === Porquê Julia?

Reviewer: ```suggestion
=== Por que Julia?
```

> O Julia ((("Julia", "about"))) foi originalmente lançada em 2012 por Alan Edelman, Stefan Karpinski, Jeff Bezanson, e Viral Shah. Ela é uma linguagem de programação gratuita e open source.

Reviewer: ```suggestion
O Julia ((("Julia", "sobre"))) foi originalmente lançada em 2012 por Alan Edelman, Stefan Karpinski, Jeff Bezanson, e Viral Shah. Ela é uma linguagem de programação gratuita e open source.
```

Escolher uma linguagem de programação é sempre subjetivo. Para mim, as seguintes características do Julia são decisivas:

- O Julia é desenvolvida como uma linguagem de programação de alta performance.
- O Julia usa despacho múltiplo, que permite ao programador escolher diferentes padrões de programação adaptados à aplicação.
- O Julia é uma linguagem dinamicamente tipada que pode ser facilmente usada interativamente.
- O Julia possui uma boa sintaxe de alto nível que é fácil de aprender.
- O Julia é uma linguagem de programação de tipo opcional, cujos tipos de dados (definidos pelo usuário) tornam o código mais claro e robusto.
- O Julia possui uma biblioteca padrão extendida e numerosos pacotes de terceiros que estão disponíveis.

> O Julia é uma linguagem de programação única pois resolve o problema chamado de "problema de duas linguagens". Nenhuma outra linguagem de programação é necessária para escrever código de alta performance. Isso não significa que isso acontece automaticamente. É de responsabilidade do programador otimizar o código que forma um gargalo, mas isso pode ser feito no próprio Julia.

Reviewer: ```suggestion
O Julia é uma linguagem de programação única pois resolve o chamado "problema de duas linguagens". Nenhuma outra linguagem de programação é necessária para escrever código de alta performance. Isso não significa que isso acontece automaticamente. É de responsabilidade do programador otimizar o código que forma um gargalo, mas isso pode ser feito no próprio Julia.
```

=== Para Quem é Este Livro?

Este livro é para qualquer um que quer aprender a programar. Nenhum conhecimento prévio é requerido.

Novos conceitos são introduzidos gradualmente e tópicos mais avançados são descritos em capítulos posteriores.

> _Think Julia_ pode ser usado para um curso de um semestre em nível de ensino médio ou de universidade.

Reviewer: ```suggestion
_Introdução à programação em Julia_ (_Think Julia_) pode ser usado para um curso de um semestre em nível de ensino médio ou de universidade.
```

=== Convenções Adotadas Neste Livro

> As ((("conventions used in this book")))((("typographical conventions in this book"))) seguintes conveções foram adotadas neste livro:

Reviewer: ```suggestion
As ((("convenções usadas neste livro")))((("convenções tipográficas deste livro"))) seguintes conveções foram adotadas neste livro:
```

_Itálico_:: indica novos termos, URLs, endereços de email, nomes de arquivo e extensões de arquivo.

+Comprimento constante+:: Usado para listagens de programas, assim como dentro de paragráfos para referir a elementos de programas como variáveis ou nomes de função, banco de dados, tipos de dado, variáveis de ambiente, declarações e palavras-chave.

**`Comprimento constante em negrito`**:: Indica comandos ou outro texto que deve ser literalmente digitado pelo usuário.

_++Comprimento constante itálico++_:: Indica texto que deve ser substituido por valores fornecidos pelo usuário ou por valores determinados pelo contexto.

[TIP]
====

> Este ((("icons used in this book")))elemento signfica uma dica ou sugestão.

Reviewer: ```suggestion
Este ((("ícones usados neste livro")))elemento signfica uma dica ou sugestão.
```

====

[NOTE]
====
Este elemento significa uma observação geral.
====

[WARNING]
====
Este elemento indica um aviso ou cuidado.
====

=== Usando Exemplos de Código

> Todo ((("code examples in this book")))((("Git")))((("GitHub")))((("repository")))((("online resources", "Julia")))código usado neste livro está disponível em um repositório Git no Github: https://github.com/JuliaIntro/JuliaIntroBR.jl. Se você não esta famialiarizado com o Git, ele é um sistema de controle de versão que permite ao usuário acompanhar os arquivos que fazem parte de um projeto. Um conjunto de arquivos sob controle do Git é chamado de “repositório”. O Github é um serviço de hospedamento que fornece armazenamento para repositórios Git e uma interface web conveniente.

Reviewer: ```suggestion
Todo ((("exemplos de código neste livro")))((("Git")))((("GitHub")))((("repositório")))((("recursos online", "Julia")))código usado neste livro está disponível em um repositório Git no Github: https://github.com/JuliaIntro/JuliaIntroBR.jl. Se você não esta familiarizado com o Git, ele é um sistema de controle de versão que permite ao usuário acompanhar os arquivos que fazem parte de um projeto. Um conjunto de arquivos sob controle do Git é chamado de “repositório”. O Github é um serviço de hospedamento que fornece armazenamento para repositórios Git e uma interface web conveniente.
```

> Um ((("packages", "installing")))((("add command, in REPL")))pacote de conveniência é fornecido que pode ser diretamente adicionado ao Julia. Apenas digite pass:[<code>add <a https://github.com/JuliaIntro/JuliaIntroBR.jl">https://github.com/JuliaIntro/JuliaIntroBR.jl</a></code>] No REPL em modo Pkg, veja <<turtles>>.

Reviewer: ```suggestion
Um ((("pacotes", "instalando")))((("adicionar comando, no REPL")))pacote de conveniência é fornecido que pode ser diretamente adicionado ao Julia. Apenas digite pass:[<code>add <a https://github.com/JuliaIntro/JuliaIntroBR.jl">https://github.com/JuliaIntro/JuliaIntroBR.jl</a></code>] No REPL em modo Pkg, veja <<turtles>>.
```

> A ((("Julia", "running"))) maneira mais fácil de executar código em Julia é indo para https://juliabox.com e iniciar uma sessão gratuita. Tanto o REPL quanto a interface de notebook estão disponíveis. Se ((("Julia", "installing")))você quiser ter o Julia instalado localmente no seu computador, você pode fazer o download https://juliacomputing.com/products/juliapro.html[JuliaPro] gratuitamente de Julia Computing. Ele consiste de uma versão recente do Julia, um ambiente interativo de desenvolvimento baseado no Atom e um número de pacotes Julia pré-instalados. Se você quiser se aventurar ainda mais, você pode fazer o download do Julia em https://julialang.org, instalar o editor que você quiser (ex. Atom ou Visual Studio Code), e ativar plug-ins para integração do Julia. Para ((("IJulia package")))uma instalação local, você também pode adicionar o pacote +IJulia+ e executar um notebook Jupyter no seu computador.

Reviewer: ```suggestion
A ((("Julia", "rodando"))) maneira mais fácil de executar código em Julia é indo para https://repl.it e iniciar uma sessão gratuita. Tanto o REPL quanto um editor estão disponíveis. Se ((("Julia", "instalando")))você quiser ter o Julia instalado localmente no seu computador, você pode fazer o download do https://juliacomputing.com/products/juliapro.html[JuliaPro] gratuitamente de Julia Computing. Ele consiste de uma versão recente do Julia, um ambiente interativo de desenvolvimento baseado no Atom e um número de pacotes Julia pré-instalados. Se você quiser se aventurar ainda mais, você pode fazer o download do Julia em https://julialang.org, instalar o editor que você quiser (ex. Atom ou Visual Studio Code), e ativar plug-ins para integração do Julia. Para ((("pacote IJulia")))uma instalação local, você também pode adicionar o pacote +IJulia+ e executar um notebook Jupyter no seu computador.
```

=== Agradecimentos

Eu realmente gostaria de agradecer ao Allen por escrever _Think Python_ e permitir que eu pudesse adaptar o seu livro para o Julia. Seu entusiasmo é contagiante!

> Eu também gostaria de agradecer aos revisores técnicos para este livro, que fizeram muitas sugestões utéis: Tim Besard, Bard Janssens, e David P. Sanders.

Reviewer: ```suggestion
Eu também gostaria de agradecer aos revisores técnicos para este livro, que fizeram muitas sugestões utéis: Tim Besard, Bart Janssens, e David P. Sanders.
```

Obrigado à Melissa Potter da O'Reilly Media, que fez deste livro melhor. Você me forçou a fazer as coisas certo e fazer este livro o mais original possível.

Obrigado a Matt Hacker da O'Reilly Media, que me ajudou com o conjunto de ferramentas Atlas e alguns problemas de destacação de sintaxe.

Obrigado a todos os estudantes que trabalharam com uma versão inicial deste livro e todos os contribuidores (listados abaixo) que mandaram correções e sugestões.

=== Lista de Contribuidores

> Se você tem uma sugestão ou correção, porfavor mande um email à ben.lauwens@gmail.com ou abra um issue em https://github.com/JuliaIntro/JuliaIntroBR.jl[GitHub]. Se eu fizer uma mudança baseada no seu feedback, irei adicioná-lo na lista de contribuidores (a menos que você peça para ser omitido)

Reviewer: ```suggestion
Se você tem uma sugestão ou correção, porfavor mande um email à ben.lauwens@gmail.com ou abra um issue em https://github.com/BenLauwens/ThinkJulia.jl[GitHub]. Se eu fizer uma mudança baseada no seu feedback, irei adicioná-lo na lista de contribuidores (a menos que você peça para ser omitido).
```
